src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_suite_test.go

```diff
@@ -42,31 +43,24 @@ var (
 	eLock          *sync.Mutex
 	httpClient     *http.Client
 	consulRunner   *consulrunner.ClusterRunner
+	setupParams    arguments
 )
 
 func TestMetricsCollector(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "MetricsCollector Suite")
 }
 
+type arguments struct {
+	McBuildPath  string
+	CcNoaaUaaURL string
+	Lock         *sync.Mutex
+}
+
 var _ = SynchronizedBeforeSuite(func() []byte {
 	mc, err := gexec.Build("autoscaler/metricscollector/cmd/metricscollector", "-race")
 	Expect(err).NotTo(HaveOccurred())
 
-	return []byte(mc)
-}, func(pathsByte []byte) {
-
-	consulRunner = consulrunner.NewClusterRunner(
-		consulrunner.ClusterRunnerConfig{
-			StartingPort: 9001 + GinkgoParallelNode()*consulrunner.PortOffsetLength,
-			NumNodes:     1,
-			Scheme:       "http",
-		},
-	)
-	consulRunner.Start()
-	consulRunner.WaitUntilReady()
-	mcPath = string(pathsByte)
-
 	ccNOAAUAA = ghttp.NewServer()
```

Comment:
I missed the "loading all app IDs" part of metrics collector.  While we can setup database in primary node with only one app in policy_json and each node has its own ccNOAAUAA sever.   The main reason is my comments below on the eLock, it can not be shared among nodes by marshall and unmarshall.